server/re/src/irods_re_serialization.cpp

```diff
@@ -1199,6 +1199,77 @@ namespace irods::re_serialization
         }
     } // serialize_vector_of_strings_ptr
 
+    static irods::error serialize_execMyRuleInp_ptr(boost::any _p, serialized_parameter_t& _out)
+    {
+        try {
+            log_re::trace(__func__);
+            const auto* casted_ptr = boost::any_cast<execMyRuleInp_t*>(_p);
+            if(!casted_ptr) {
+                _out["execMyRuleInp_ptr"] = "nullptr";
+                return SUCCESS();
+            }
+            _out["myRule"] = casted_ptr->myRule;
+            _out["addr_hostAddr"] = casted_ptr->addr.hostAddr;
+            _out["addr_zoneName"] = casted_ptr->addr.zoneName;
+            _out["addr_portNum"] = boost::lexical_cast<std::string>(casted_ptr->addr.portNum);
+            _out["addr_dummyInt"] = boost::lexical_cast<std::string>(casted_ptr->addr.dummyInt);
+            serialize_keyValPair(casted_ptr->condInput, _out);
+            _out["outParamDesc"] = casted_ptr->outParamDesc;
```

Comment:
I'm guessing none of these need to be checked for null because they are fixed size buffers (no allocation necessary)?

Reply:
I believe so. In order...
`myRule` is `char[META_STR_LEN]`
`addr` is `rodsHostAddr_t` (not a pointer)
`hostAddr` is `char[LONG_NAME_LEN]`
`zoneName` is `char[NAME_LEN]`
`portNum` and `dummyInt` are `int`
`outParamDesc` is `char[LONG_NAME_LEN]`